packages/tupaia-web/src/pages/ModalRoute.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { MODAL_TYPES } from '../constants';
+import { Projects } from '.';
+import { Modal } from '../components';
+
+/**
+ * This is the wrapper to handle any search param routes that should be modals
+ */
+export const ModalRoute = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const modal = searchParams.get('modal') as typeof MODAL_TYPES[keyof typeof MODAL_TYPES];
+  // If no modal param or invalid modal param, return null
+  if (!modal || !Object.values(MODAL_TYPES).includes(modal)) return null;
+
+  const modalViews = {
```

Comment:
I don't feel too strongly but have a slight preference for putting the modalViews const outside the component body. Just above the component in this file. I think it's a bit easier to reference that way.